.github/workflows/macM1-e2e.yaml

```diff
@@ -0,0 +1,58 @@
+name: e2e tests on Mac M1
+
+on:
+  workflow_dispatch:
+  schedule:
+  - cron: '0 9 * * 1-5'
+
+
+jobs:
+
+  e2e-tests:
+    timeout-minutes: 45
+    runs-on: [self-hosted, macos-latest, arm64]
+    env:
+      M1: 1
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          ref: main
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '^1.18'
+      - name: Install dependencies
+        run: npm ci
+        #The next 2 steps are a workaround for an unexpected failure in launching electron before running e2e tests
+        #Such failure is addressed in issue #2915
+      - name: add tools to PATH
+        run: |
+          mkdir -p ~/.rd/bin
+          cp -R ~/actions-runner/_work/rancher-desktop/rancher-desktop/resources/darwin/bin ~/.rd/bin
+          export PATH="~/.rd/bin:$PATH"
+      - name: Run Rancher Desktop in dev
+        run: |
+          npm run dev --no-modal-dialogs &
+          sleep 300
+          ~/.rd/bin/rdctl shutdown
+          wait
+      - name: Run e2e Tests
+        continue-on-error: false
+        run: npm run test:e2e
+      - name: Failed tests
+        if: failure()
+        run: mkdir -p ./e2e/reports
```

Comment:
As mentioned in the last round of comments, I don't think this is necessary (because if it's missing, upload artifacts would just emit a warning… and the build would have already failed).  But at least it's harmless.

The name ("Failed tests") is not very descriptive of the action, though.